File: src/main/java/dev/swe573/whatsthis/dto/PostDto.java
package dev.swe573.whatsthis.dto;

import lombok.Data;

import java.util.List;
import java.time.LocalDateTime;

@Data
public class PostDto {
    private Long id;
    private String title;
    private String description;
    private int votes;
    private Long userId;

    private List<String> tags;
    private List<String> imageUrls;

    private String material;
    private String size;
    private String textAndLanguage;
    private String color;
    private String shape;
    private String weight;
    private String price;
    private String location;
    private String timePeriod;
    private String smell;
    private String taste;
    private String texture;
    private String hardness;
    private String pattern;
    private String brand;
    private String print;
    private String icons;
    private Boolean handmade;
    private String functionality;

    private String widthValue;
    private String widthUnit;
    private String heightValue;
    private String heightUnit;
    private String depthValue;
    private String depthUnit;

    private List<CommentDto> comments;
    private List<PartDto> parts;

    private Long solutionCommentId;
    private boolean isSolved;

    private LocalDateTime createdAt;

}

File: mobile/eurekaNexus/app/src/main/java/com/yusuf/eurekanexus/data/model/PostDto.kt
package com.yusuf.eurekanexus.data.model

import com.google.gson.annotations.SerializedName

data class PostDto(
    @SerializedName("title") val title: String,
    @SerializedName("description") val description: String,
    @SerializedName("votes") val votes: Int,
    @SerializedName("userId") val userId: Long,
    @SerializedName("tags") val tags: List<String>,
    @SerializedName("material") val material: String?,
    @SerializedName("size") val size: String?,
    @SerializedName("textAndLanguage") val textAndLanguage: String?,
    @SerializedName("color") val color: String?,
    @SerializedName("shape") val shape: String?,
    @SerializedName("weight") val weight: String?,
    @SerializedName("price") val price: String?,
    @SerializedName("location") val location: String?,
    @SerializedName("timePeriod") val timePeriod: String?,
    @SerializedName("smell") val smell: String?,
    @SerializedName("taste") val taste: String?,
    @SerializedName("texture") val texture: String?,
    @SerializedName("hardness") val hardness: String?,
    @SerializedName("pattern") val pattern: String?,
    @SerializedName("brand") val brand: String?,
    @SerializedName("print") val print: String?,
    @SerializedName("handmade") val handmade: Boolean?,
    @SerializedName("functionality") val functionality: String?,
    @SerializedName("widthValue") val widthValue: String?,
    @SerializedName("widthUnit") val widthUnit: String?,
    @SerializedName("heightValue") val heightValue: String?,
    @SerializedName("heightUnit") val heightUnit: String?,
    @SerializedName("depthValue") val depthValue: String?,
    @SerializedName("depthUnit") val depthUnit: String?,
    @SerializedName("comments") val comments: List<CommentDto>?,
    @SerializedName("solutionCommentId") val solutionCommentId: Long?,
    @SerializedName("createdAt") val createdAt: String?
)